cpp/src/lance/format/schema.cc

```diff
@@ -327,6 +323,14 @@ std::shared_ptr<::arrow::DataType> Field::type() const {
   }
 }
 
+std::shared_ptr<::arrow::DataType> Field::storage_type() const {
```

Comment:
how come in `type()` we expand nested types but not in `storage_type()` ?

Reply:
e.g., if you have a nested type with extension types in subfields, `storage_type()` won't actually convert the subfield extension types right?

Reply:
Good point. will fix it.